pkg/webhookconfig/configmanager.go

```diff
@@ -685,6 +685,7 @@ func (m *webhookConfigManager) updateStatus(policy *kyverno.ClusterPolicy, statu
 	policyCopy.Status.Autogen.Requested = requested
 	policyCopy.Status.Autogen.Supported = supported
 	policyCopy.Status.Autogen.Activated = activated
+	policyCopy.Status.Rules = policy.Spec.Rules
```

Comment:
Why add the entire rule to status?

Reply:
What's the purpose of adding original (non-autogen) rules to status? While applying autogen rules, do we need to skip original ones internally?

Reply:
Yes, we can use the rules from the status directly as they contain all effective rules. It will make things simpler straightforward. Spec rules are for the policy definition expressed by the end user, Status rules are for the internal engine processing.

Reply:
ok, adding my thoughts for the implementation:

1. `Policy.GetRules()` needs to return `status.rules`, and internally Kyverno should apply these rules to resources.
2. We need to make sure `status.rules` are always up-to-date with `spec.rules`, and properly handle errors while updating `status.rules`, maybe we can set `status.conditions.ready` to `false` if error occurs.